### PR TITLE
Feature/general improvements

### PR DESCRIPTION
**Pull request by romaninsh**
This PR implements several general fixes, such as improving relationship between model and persistence drivers. This PR is necessary before Join implementation.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": ">=5.5.0",
         "atk4/dsql": "*",
-        "atk4/core": "1.0"
+        "atk4/core": "*"
```
Comment (skondakov) [inline composer.json]: So it's no more forked with specific version? Is this ok?

File: docs/hooks.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+.. important::
+
+Please never use ``$this`` inside your hook to refer to the model. The model
+is always passed as a first argument. If you ever use ``$this`` then your
+model will perform very weirdly when cloned::
+
+    $m->addHook('beforeSave', function($m) {
+        $this['name'] = 'John';  // WRONG!
+        $m['surname'] = 'Smith'; // GOOD
```
Comment (skondakov) [inline docs/hooks.rst]: Should be $m['name'] for the purpose of the example I think

**Comment by skondakov** [comment]
Seems good to me overall, just check the comments please and you can push.

File: composer.json (hunk 1 above)
Comment (romaninsh) [inline composer.json]: we need 1.0.1 version, which is now stable. Will keep both repos in sync.

File: docs/hooks.rst (hunk 2 above)
Comment (romaninsh) [inline docs/hooks.rst]: i re-checked and it actually is correct with the surname.

**Pull request merged**